### PR TITLE
Update llvm-dialects and remove dialect guards

### PR DESCRIPTION
**Pull request by nhaehnle**
The update to llvm-dialects makes the dialect guards unnecessary.

### CODE REVIEW

File: llpc/context/llpcCompiler.cpp [hunk 1]
```diff
@@ -2549,8 +2544,6 @@ Result Compiler::buildRayTracingPipelineInternal(RayTracingContext &rtContext,
     helperThreadProvider->SetTasks(&helperThreadBuildRayTracingPipelineElf, newModules.size(),
                                    static_cast<void *>(&helperThreadPayload));
     {
```
Comment (jasilvanus) [inline llpc/context/llpcCompiler.cpp:2546]: This scope looks like it was only intended for the guard. Also for some of the other guards.

**Review by jasilvanus — COMMENTED** [review]
Nice!

We may want to remove some of the scopes as well, see the inline comment.

File: llpc/context/llpcCompiler.cpp (hunk 1 above)
Comment (nhaehnle) [reply llpc/context/llpcCompiler.cpp]: Good point.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 091a2b214652736a4c086693e92a9ce8ef122299
<details><summary>CTS tests (Failed: 1/282021)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39248/70506 (55.7%)</li>
<li>Failed: 0/70506 (0.0%)</li>
<li>Not Supported: 31258/70506 (44.3%)</li><li>Warnings: 0/70506 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39497/70506 (56.0%)</li>
<li><details><summary>Failed: 1/70506 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.object_management.max_concurrent.graphics_pipeline
Stack trace: Timeout
```
</p></details></li>

<li>Not Supported: 31008/70506 (44.0%)</li><li>Warnings: 0/70506 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 36345/70508 (51.5%)</li>
<li>Failed: 0/70508 (0.0%)</li>
<li>Not Supported: 34163/70508 (48.5%)</li><li>Warnings: 0/70508 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 36247/70501 (51.4%)</li>
<li>Failed: 0/70501 (0.0%)</li>
<li>Not Supported: 34254/70501 (48.6%)</li><li>Warnings: 0/70501 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by qiaojbao** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 091a2b214652736a4c086693e92a9ce8ef122299
<details><summary>CTS tests (Failed: 0/211515)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39498/70506 (56.0%)</li>
<li>Failed: 0/70506 (0.0%)</li>
<li>Not Supported: 31008/70506 (44.0%)</li><li>Warnings: 0/70506 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 36345/70508 (51.5%)</li>
<li>Failed: 0/70508 (0.0%)</li>
<li>Not Supported: 34163/70508 (48.5%)</li><li>Warnings: 0/70508 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 36247/70501 (51.4%)</li>
<li>Failed: 0/70501 (0.0%)</li>
<li>Not Supported: 34254/70501 (48.6%)</li><li>Warnings: 0/70501 (0.0%)</li>
</ul></details>


</ul></details>

**Review by jasilvanus — APPROVED** [review]
Thanks, LGTM

**Review by amdrexu — APPROVED** [review]

**Comment by nhaehnle** [comment]
Yeah, there was an issue in the test which changes optimization levels while using the same llpc::Context, which causes the LgcContext to be repeatedly destroyed and re-created. Since the LgcContext owned the DialectContext, that caused a subtle problem.

That could be fixed, but logically the DialectContext is really quite tied to the LLVMContext. If/when we start getting dialect features upstream, I expect them to be tied to the LLVMContext as well. So I've decided to fix it by always combining the LLVMContext creation with the DialectContext creation.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6aa238be5c8cf58a8f8b1d36d86b316024bc37e9
<details><summary>CTS tests (Failed: 802/211520)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39248/70506 (55.7%)</li>
<li>Failed: 0/70506 (0.0%)</li>
<li>Not Supported: 31258/70506 (44.3%)</li><li>Warnings: 0/70506 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39498/70506 (56.0%)</li>
<li>Failed: 0/70506 (0.0%)</li>
<li>Not Supported: 31008/70506 (44.0%)</li><li>Warnings: 0/70506 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35557/70508 (50.4%)</li>
<li><details><summary>Failed: 802/70508 (1.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.buffer.suballocation.storage_texel.uniform.index.vertex.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_src.transfer_dst.storage_texel.uniform.storage.index.vertex.indirect.create.binding
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_src.uniform_texel.storage_texel.uniform.storage.indirect.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.uniform_texel.vertex.create.binding
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer_marker.graphics.external_host_mem.bottom_of_pipe.memory_dep.draw
Stack trace: Flake

FAILURE: dEQP-VK.api.command_buffers.recording_to_invalid
Stack trace: Flake

FAILURE: dEQP-VK.api.command_buffers.reset_implicit
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.blit_image.all_formats.color.2d.r4g4_unorm_pack8.a2b10g10r10_unorm_pack32.optimal_linear_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.image_to_image.all_formats.color.2d.r32_uint.r32_sfloat.general_general
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.blit_image.all_formats.color.2d.a8b8g8r8_snorm_pack32.b8g8r8a8_unorm.general_general_linear
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.blit_image.all_formats.color.2d.r4g4_unorm_pack8.a1r5g5b5_unorm_pack16.optimal_linear_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.blit_image.all_formats.generate_mipmaps.from_previous_level.layercount_1.r8g8b8a8_sint.optimal_optimal_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.buffer_to_depthstencil.buffer_offset_d16_unorm_s8_uint_S
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.image_to_image.all_formats.color.2d.b5g6r5_unorm_pack16.r16_snorm.general_optimal
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.image_to_image.all_formats.color.2d.r8g8_unorm.r16_snorm.optimal_general
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.image_to_image.all_formats.color.2d.r8g8b8a8_snorm.b8g8r8a8_unorm.general_optimal
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.resolve_image.diff_image_size.dst_266_256_1_2_bit
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.simple_tests.mirror_subregions_3d.linear
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.image_to_image.all_formats.color.1d.r8_srgb.r8_srgb.optimal_general
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.image_to_image.all_formats.color.2d.b8g8r8a8_snorm.r32_sfloat.general_optimal
Stack trace: Flake
...
```
</p></details></li>

<li>Not Supported: 34149/70508 (48.4%)</li><li>Warnings: 0/70508 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by qiaojbao** [comment]
retest this please

**Comment by qiaojbao** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 9137cb960cb0b71ec9322c6cfb15c20441be8117
<details><summary>CTS tests (Failed: 661/211487)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39248/70495 (55.7%)</li>
<li>Failed: 0/70495 (0.0%)</li>
<li>Not Supported: 31247/70495 (44.3%)</li><li>Warnings: 0/70495 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39498/70495 (56.0%)</li>
<li>Failed: 0/70495 (0.0%)</li>
<li>Not Supported: 30997/70495 (44.0%)</li><li>Warnings: 0/70495 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35697/70497 (50.6%)</li>
<li><details><summary>Failed: 661/70497 (0.9%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.buffer.suballocation.storage_texel.uniform.vertex.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_src.storage.index.vertex.indirect.create.binding_residency
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_src.uniform_texel.uniform.create.binding
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.blit_image.all_formats.color.3d.r16_sfloat.r16_sfloat.optimal_optimal_linear_stripes_y
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.buffer_to_depthstencil.buffer_offset_d16_unorm_s8_uint_D_S
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.image_to_image.all_formats.color.2d.a8b8g8r8_srgb_pack32.r32_sfloat.general_optimal
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.blit_image.all_formats.color.2d.r8g8b8a8_sint.r32_sint.general_general_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.blit_image.simple_tests.scaling_and_offset.r32_sfloat_linear
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.buffer_to_depthstencil.d32_sfloat_s8_uint_S_D
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.buffer_to_image.regions
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.image_to_image.all_formats.color.2d.bc2_srgb_block.bc7_unorm_block.general_general
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.all_formats.color.1d.a2r10g10b10_unorm_pack32.a2r10g10b10_unorm_pack32.optimal_general_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.all_formats.color.2d.e5b9g9r9_ufloat_pack32.b5g6r5_unorm_pack16.general_general_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.all_formats.color.2d.r4g4_unorm_pack8.r16g16_snorm.optimal_linear_linear
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.all_formats.color.2d.r8g8b8a8_uint.r8_uint.general_optimal_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.all_formats.generate_mipmaps.from_base_level.layercount_1.r8g8_unorm.general_optimal_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.all_formats.generate_mipmaps.from_previous_level.layercount_6.r16g16_snorm.general_general_linear
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.all_formats.generate_mipmaps.from_previous_level.layercount_6.r8g8b8a8_snorm.general_general_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.buffer_to_depthstencil.d16_unorm_s8_uint_SD
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.image_to_image.all_formats.depth_stencil.2d.d32_sfloat_s8_uint_d32_sfloat_s8_uint_depth_stencil_aspects.general_optimal
Stack trace: Flake
...
```
</p></details></li>

<li>Not Supported: 34139/70497 (48.4%)</li><li>Warnings: 0/70497 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by qiaojbao** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 9137cb960cb0b71ec9322c6cfb15c20441be8117
<details><summary>CTS tests (Failed: 0/211487)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39248/70495 (55.7%)</li>
<li>Failed: 0/70495 (0.0%)</li>
<li>Not Supported: 31247/70495 (44.3%)</li><li>Warnings: 0/70495 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39498/70495 (56.0%)</li>
<li>Failed: 0/70495 (0.0%)</li>
<li>Not Supported: 30997/70495 (44.0%)</li><li>Warnings: 0/70495 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36345/70497 (51.6%)</li>
<li>Failed: 0/70497 (0.0%)</li>
<li>Not Supported: 34152/70497 (48.4%)</li><li>Warnings: 0/70497 (0.0%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]

**Pull request merged**